python312Packages.meshtastic: 2.5.5 -> 2.5.6post1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.meshtastic is: Python API for talking to Meshtastic devices

meta.homepage for python312Packages.meshtastic is: https://github.com/meshtastic/Meshtastic-python

meta.changelog for python312Packages.meshtastic is: https://github.com/meshtastic/python/releases/tag/2.5.6post1


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 2.5.6post1 in filename of file in /nix/store/w3nzfjfzf53z55vax5yzl3fwqjda3v7v-python3.12-meshtastic-2.5.6post1

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
3 total rebuild path(s)

3 package rebuild(s)

First fifty rebuilds by attrpath

python311Packages.meshtastic
python312Packages.meshtastic
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/w3nzfjfzf53z55vax5yzl3fwqjda3v7v-python3.12-meshtastic-2.5.6post1 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.meshtastic https://github.com/r-ryantm/nixpkgs/archive/a693796eb0c2bf350a2c5932d4d7361a8f001a6e.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/a693796eb0c2bf350a2c5932d4d7361a8f001a6e#python312Packages.meshtastic
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/w3nzfjfzf53z55vax5yzl3fwqjda3v7v-python3.12-meshtastic-2.5.6post1
ls -la /nix/store/w3nzfjfzf53z55vax5yzl3fwqjda3v7v-python3.12-meshtastic-2.5.6post1/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.meshtastic</li>
    <li>python311Packages.meshtastic.dist</li>
    <li>python312Packages.meshtastic</li>
    <li>python312Packages.meshtastic.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @fabaff for [testing](https://github.com/ryantm/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
python312Packages.meshtastic: 2.5.5 -> 2.5.6post1

NickCao reviewed (commented):
A bunch of dependencies are dropped.

fabaff committed:
python312Packages.meshtastic: 2.5.6post1 -> 2.5.9

Diff: https://github.com/meshtastic/Meshtastic-python/compare/None...2.5.9

Changelog: https://github.com/meshtastic/python/releases/tag/2.5.9

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 367038`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.meshtastic</li>
    <li>python312Packages.meshtastic.dist</li>
  </ul>
</details>
